google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClientFactory.java

```diff
@@ -136,9 +140,16 @@ public BigtableDataClient createForAppProfile(@Nonnull String appProfileId) thro
     BigtableDataSettings.Builder settingsBuilder =
         defaultSettings.toBuilder().setAppProfileId(appProfileId);
```

Comment:
nit. you dont need to modify this after this point, so might as well build it here as opposed to building the stubSettings & Client settings separately